File: 27-RemoveElement/27-RemoveElement.cpp
// Last updated: 16/10/2025, 04:31:14
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
     int cnt = 0;
     int n = nums.size();
     for(auto it = nums.begin(); it != nums.end();){
         if(*it == val){
             nums.erase(it);
         }else{
             cnt++;
             it++;
         }
     }
        return cnt;
    }
};

File: 27-RemoveElement/27-RemoveElement.java
// Last updated: 16/10/2025, 04:31:07
class Solution {
    public int removeElement(int[] nums, int val) {
        int n = nums.length;
        int cur = 0;
        int last = 0;

        while(cur < n){
            if(nums[cur] != val){
                nums[last++] = nums[cur];
            }
            cur++;
        }
        return last;
    }
}